packages/wp-story-editor/src/api/media.js

```diff
@@ -85,13 +85,35 @@ export function getMediaById(config, mediaId) {
   return apiFetch({ path }).then(getResourceFromAttachment);
 }
 
+/**
+ * Get the optimized variant of a given media resource.
+ *
+ * @param {Object} config Configuration object.
+ * @param {number} mediaId Media ID.
+ * @return {Promise<import('@web-stories-wp/media').Resource>} Media resource if found, null otherwise.
+ */
+export async function getOptimizedMediaById(config, mediaId) {
+  const path = addQueryArgs(`${config.api.media}${mediaId}/`, {
+    context: 'edit',
+    _fields: 'meta.web_stories_optimized_id',
+  });
+
+  const result = await apiFetch({ path });
+
+  if (result?.meta?.web_stories_optimized_id) {
+    return getMediaById(config, result?.meta?.web_stories_optimized_id);
```

Comment:
```suggestion
    return getMediaById(config, result.meta.web_stories_optimized_id);
```
Not needed as the line above does this check.